src/views/Alteration.vue

```diff
@@ -99,42 +99,43 @@ export default class Alteration extends Mixins(LegalApiMixin, FilingTemplateMixi
       return
     }
 
-    // do not proceed if we don't have the necessary query params
-    // if (isNaN(this.filingId)) {
-    //   const err = 'Invalid filing ID'
-    //   console.log(err) // eslint-disable-line no-console
-    //   this.emitFetchError(err)
-    //   return
-    // }
-
     // try to fetch data
     try {
-      // fetch IA to alter
-      const { filing } = await this.fetchFilingByType(this.INCORPORATION_APPLICATION)
-
-      // do not proceed if this isn't an IA filing
-      if (!filing.incorporationApplication) {
-        throw new Error('Invalid IA filing')
-      }
-
-      // do not proceed if this isn't a COMPLETED filing
-      if (filing.header.status !== FilingStatus.COMPLETED) {
-        throw new Error('Invalid IA status')
-      }
-
-      // parse IA filing into store
-      this.parseAlteration(filing)
-
       // set current entity type
       this.setEntityType(EntityTypes.BCOMP)
 
       // initialize Fee Summary data
+      // TODO: Set/Clear Data according to filing type / entity type
       this.setFilingData({
         filingTypeCode: FilingCodes.ALTERATION,
         entityType: EntityTypes.BCOMP,
         priority: false
       })
 
+      if (this.alterationId) {
```

Comment:
I just updated this similarly to Correction.vue. Should make our job easier in the future :)